languages/swift/exercises/concept/enums/.docs/after.md

```diff
@@ -0,0 +1,284 @@
+
+```swift
+enum NESButton: Character {
+  case up = "⬆️"
+  case down = "⬇️"
+  case left = "⬅️"
+  case right = "➡️"
+  case a = "🅰️"
+  case b = "🅱️"
+  case select = "✅"
+  case start = "🚦"
+}
+```
+
+Here, the `: Character` tells the compiler that the raw values will be of type `Character` and the assignment of the raw value follows each value.
+
+Raw values can be accessed through each value's automatically generated `rawValue` property.
+
+```swift
+NESButton.left.rawValue
+// => "⬅️"
+NESButton.b.rawValue
+// => "🅱️"
+```
+
+Raw values can also be used to initialize values of the enum type. Note that the return value is an optional, as an invalid raw value may be passed in.
+
+```swift
+let upButton = NESButton.init(rawValue: "⬆️")
+// => Optional(.up)
+let invalid = NESButton.init(rawValue: "🙄")
+// => nil
+```
+
+Swift can implicitly assign raw values for `String` and `Int` raw values. If a raw value type of `String` is specified, the raw value will be implicitly assigned to the name on the value as a String unless the implicit value is overridden with an explicit assignment.
+
+If a raw value type of `Int` is specified, the raw value will be implicitly assigned to an int 1 greater than the previous case's raw value, unless overridden with an explicit assignment. The default raw value for the first case is 0 unless otherwise specified.
+
+```swift
+enum Coin: String {
+  case heads
+  case tails = "eagle"
+}
+
+Coin.head.rawValue
+// => "heads"
+Coin.tails.rawValue
+// => "eagle"
+
+enum Dwarf: Int {
+  case grumpy, sleepy, sneezy, happy = 8, bashful, dopey, doc
+}
+
+Dwarf.grumpy.rawValue
+// => 0
+Dwarf.sneezy.rawValue
+// => 2
+Dwarf.happy.rawValue
+// => 8
+Dwarf.bashful.rawValue
+// => 9
+```
+
+### Associated values
+
+Enums in Swift can also carry another type of information known as [_associated values_][associated-values]. With associated values, each case of the enum may have a different type of value that can be carried along with the enum value. The types of the associated values are specified in the enum declaration.
+
+```swift
+enum Suit: Character {
+  case diamonds = "♦️", clubs = "♣️", hearts = "♥️", spades = "♠️"
+}
+
+enum Card {
+  case ace(Suit), king(Suit), queen(Suit), jack(Suit)
+  case number(Suit, Int)
+}
+
+let aceOfSpades = Card.ace(.spades)
+let twoOfHearts = Card.number(.hearts, 2)
+```
+
+These values can then be extracted and used via pattern matching.
+
+```swift
+func printCard(_ card: Card) {
+  switch card {
+  case .ace(let suit): print("A\(suit.rawValue)")
+  case .king(let suit): print("K\(suit.rawValue)")
+  case .queen(let suit): print("Q\(suit.rawValue)")
+  case .jack(let suit): print("J\(suit.rawValue)")
+  case let .number(suit, value): print("\(value)\(suit.rawValue)")
+  }
+}
+
+printCard(aceOfSpades)
+// prints "A♠️"
+printCard(twoOfHearts)
+// prints "2♥️"
+```
+
+### Recursive enumerations
```

Comment:
FYI: while this could be added to this exercise, you could consider having a separate recursion exercise to touch upon this concept.